app/src/main/java/org/projectbuendia/client/models/ObsValue.java

```diff
@@ -0,0 +1,247 @@
+// uted under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+// OR CONDITIONS OF ANY KIND, either express or implied.  See the License for
+// specific language governing permissions and limitations under the License.
+
+package org.projectbuendia.client.models;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+
+import org.joda.time.Instant;
+import org.joda.time.LocalDate;
+import org.joda.time.ReadableInstant;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.projectbuendia.client.utils.Utils;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+/** The value of an observation, represented as a union of all the possible data types. */
+public final class ObsValue implements Comparable<ObsValue> {
+    /** The observed value as a concept UUID, if the observation is for a coded concept. */
+    public final @Nullable String uuid;
+
+    /** The observed numeric value, if the observation is for a numeric concept. */
+    public final @Nullable Double number;
+
+    /** The observed string value, if the observation is for a text concept. */
+    public final @Nullable String text;
+
+    /** The observed value as a LocalDate, if the observation is for a calendar date. */
+    public final @Nullable LocalDate date;
+
+    /** The observed value as an Instant, if the observation is for an instant in time. */
+    public final @Nullable Instant instant;
+
+    /**
+     * The localized name of the observed value concept (specified in the 'uuid' field).
+     * This is initially null, not used for comparison, and filled in only as needed for display.
+     */
+    public @Nullable String name;
+
+    public static final ObsValue MIN_VALUE = ObsValue.newCoded(false);
+    public static final ObsValue MAX_VALUE = ObsValue.newTime(Long.MAX_VALUE);
+    public static final ObsValue MIN_DATE = ObsValue.newDate(Utils.MIN_DATE);
+    public static final ObsValue MAX_DATE = ObsValue.newDate(Utils.MAX_DATE);
+    public static final ObsValue MIN_TIME = ObsValue.newTime(Utils.MIN_TIME);
+    public static final ObsValue MAX_TIME = ObsValue.newTime(Utils.MAX_TIME);
+    public static final ObsValue ZERO = ObsValue.newNumber(0);
+    public static final ObsValue FALSE = ObsValue.newCoded(false);
+    public static final ObsValue TRUE = ObsValue.newCoded(true);
+
+    private static final Map<String, Integer> CODED_VALUE_ORDERING =
+        new ImmutableMap.Builder<String, Integer>()
+            .put(ConceptUuids.NO_UUID, -100)
+            .put(ConceptUuids.NONE_UUID, -1)
+            .put(ConceptUuids.NORMAL_UUID, -1)
+            .put(ConceptUuids.SOLID_FOOD_UUID, -1)
+            .put(ConceptUuids.MILD_UUID, 1)
+            .put(ConceptUuids.MODERATE_UUID, 2)
+            .put(ConceptUuids.SEVERE_UUID, 3)
+            .put(ConceptUuids.YES_UUID, 100).build();
```

Comment:
Yes, I agree it would be better to define the ordering elsewhere and not have it hardcoded here.  At the moment this seems the only place to put this logic, though.  

To get it from the profile would require:
- Defining how it should be specified in the CSV profile
- Deciding where to store this information in OpenMRS, and writing the code (in buendia-profile-apply) to import it into OpenMRS
- Deciding how to expose this information via the API, and extending the Buendia OpenMRS module to provide the ordering information through the API
- Deciding where/how to store this information on the client, and writing the client code to sync the information down from the server to the client
- Modifying the client to use this information when sorting ObsValues.

So, it's a substantial piece of work.  I suggest we leave a TODO comment here to look into this possibility in the future and proceed with the hardcoded implementation for now.